src/FilterPanel.tsx

```diff
@@ -0,0 +1,100 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+
+interface FilterPanelProps {
+  setFilterMode: React.Dispatch<React.SetStateAction<string>>,
+  filteredTodos: Todo[];
+  setFilteredTodos: React.Dispatch<React.SetStateAction<Todo[]>>
+}
+
+export const FilterPanel: React.FC<FilterPanelProps> = ({
+  setFilterMode, filteredTodos, setFilteredTodos,
+}) => {
+  const [activeButton, setActiveButton] = useState<string>('All');
+
+  const hasCompletedTasks = filteredTodos.some((todo) => todo.completed);
+  const activeTodoCounter = filteredTodos.reduce((counter, todo) => {
+    if (!todo.completed) {
+      // eslint-disable-next-line no-param-reassign
+      counter += 1;
+    }
+
+    return counter;
+  }, 0);
+
+  const leftTodosText = (activeTodoCounter === 1) ? '1 item left' : `${activeTodoCounter} items left`;
+
+  const showActiveTodos = (filterName: string) => {
+    setFilterMode('Active');
+    setActiveButton(filterName);
+  };
+
+  const showAllTodos = (filterName: string) => {
+    setFilterMode('All');
+    setActiveButton(filterName);
+  };
+
+  const showCompletedTodos = (filterName: string) => {
+    setFilterMode('Completed');
+    setActiveButton(filterName);
+  };
```

Comment:
It seems that you repeat the same logic in these functions. Create a general one, please

Reply:
fixed